DEVELOPMENT.md

```diff
@@ -16,6 +16,9 @@ interactions, and other internal behavior, which leads to fragile tests.
 They also make it possible to mock complex interfaces or classes, which hides the fact that these classes are not (easily) testable. 
 We prefer to write mocks by hand, which forces code to be written in a certain testable style.
 * Use an assertion from Airlift's `Assertions` class if there is one that covers your case rather than writing the assertion by hand. Over time we may move over to more fluent assertions like AssertJ.
+* Using ``var`` is allowed.
+* Prefer using immutable collections from Guava over unmodifiable collections from JDK.
```

Comment:
Maybe add some rationale here:
```
 * Guava immutable collections preserve insertion order which is important for planning and query determinism
 * Guava ``copyOf`` does not make a copy of collection if source collection is also Guava immutable collection.
```

Reply:
>  Guava ``copyOf``

Same for JDK `of` (like `List.of`)

Reply:
> which is important for planning and query determinism

I guess they are more reasons behind this.  I mentioned only deterministic iteration.

Reply:
> Same for JDK of (like List.of)

That's true, but without changing all usages of Guava to JDK, there will be transitions between Guava <-> JDK, which would trigger full copy. So it's either none or nothing in non-testing code

Reply:
That is true. But I am not sure if this need to be document. It is kind of obvious that they should not be mixed.